packages/sites/clinepi-site/webapp/js/client/store-modules/AccessRequestStoreModule.js

```diff
@@ -169,53 +233,107 @@ function observeSubmitForm(action$, state$, dependencies) {
     filter(({ type }) => type === SUBMIT_FORM),
     withLatestFrom(state$),
     mergeMap(async ([, { [key]: accessRequestState }]) => {
-      const response = await jsonPutRequest(
-        dependencies.wdkService.serviceUrl,
-        `/users/${userId(accessRequestState)}/access-request/${datasetId(accessRequestState)}`,
-        formValues(accessRequestState)
+      const studyAccessRequestData = await getStudyAccessRequestData(
+        dependencies.studyAccessApi,
+        userId(accessRequestState),
+        datasetId(accessRequestState)
       );
```

Comment:
I think this request is being made to determine if the data access request is being updated. Instead, could we check if `requestStatus(accessRequestState) === 'denied'`?

Reply:
Ahh, good catch! When I first wired this up, I couldn't think of a way to avoid the network request. I later decided to store the access request data on load, but didn't think/remember to update this code!